src/rpm/headers/signature_builder.rs

```diff
@@ -111,21 +111,37 @@ impl SignatureHeaderBuilder<Empty> {
 
 impl SignatureHeaderBuilder<WithDigest> {
     /// add a signature over the header and a signature across header and source excluding the static lead
-    pub fn add_signature(
+    pub fn add_rsa_signature(
         mut self,
-        rsa_sig_header_only: &[u8],
-        rsa_sig_header_and_archive: &[u8],
+        sig_header_only: &[u8],
+        sig_header_and_archive: &[u8],
     ) -> SignatureHeaderBuilder<WithSignature> {
         let offset = 0i32; // filled externally later on
         self.entries.push(IndexEntry::new(
             IndexSignatureTag::RPMSIGTAG_RSA,
             offset,
-            IndexData::Bin(rsa_sig_header_only.to_vec()),
+            IndexData::Bin(sig_header_only.to_vec()),
         ));
         self.entries.push(IndexEntry::new(
             IndexSignatureTag::RPMSIGTAG_PGP,
             offset,
-            IndexData::Bin(rsa_sig_header_and_archive.to_vec()),
+            IndexData::Bin(sig_header_and_archive.to_vec()),
+        ));
+        SignatureHeaderBuilder::<WithSignature> {
+            entries: self.entries,
+            phantom: Default::default(),
+        }
+    }
+
+    pub fn add_eddsa_signature(
+        mut self,
+        sig_header_only: &[u8],
+    ) -> SignatureHeaderBuilder<WithSignature> {
+        let offset = 0i32; // filled externally later on
+        self.entries.push(IndexEntry::new(
+            IndexSignatureTag::RPMSIGTAG_DSA,
+            offset,
+            IndexData::Bin(sig_header_only.to_vec()),
```

Comment:
I am not familiar with this, but why we dont' need 'RPMSIGTAG_GPG' here?

Reply:
Because EdDSA support was introduced only a few years ago, and any package which is signed as such ought to have payload digests too, which makes checking the legacy v3 style signatures pointless.

The only reason to check v3 style signatures is that EL7 packages aren't guaranteed to have payload digests, so the only way to verify the integrity of the payload is to use the v3 header+payload style ones.

Reply:
thanks!